### DISCUSSION TITLE
Files from resources randomly moves to conf directory after update 2.8.11 -> 2.9.5

### DISCUSSION THREAD

**Discussion by kostafey** [playframework/playframework#12824] [General]
I have upgraded Play 2.8.11 -> 2.9.5 and encountered the following problem.
When I run `sbt dist`, some files (seem to be randomly selected) are moved from the `resources` directory to the `conf` directory.
Other files are still located in `resources`.

What is the reason for that, and how can I keep all resources in the `resources` directory?

**Comment by mkurz** [comment]
Which `resources` directory do you mean? `src/main/resources`?

**Reply by kostafey** [reply]
I used `/resources` from the root directory of the project. Most of the files from this directory moved to the `/conf` inside the final .zip file in `/target/universal/` created by `sbt dist`command.

Here is a part of my `build.sbt` file:

```scala
Compile / unmanagedResourceDirectories += baseDirectory.value / "resources"
Compile / unmanagedResourceDirectories += baseDirectory.value / "test/resources"
```

**Reply by mkurz** [reply]
I tried the same with a Play 2.8.22 and 3.0.5 Play app, with your described `unmanagedResourceDirectories` settings and all the files within the `resources` folder were put into the `conf` folder of the distribution, no matter if Play 2.8 or Play 3. So there is no difference and also this is totally expected. This is what the `unmanagedResourceDirectories` setting is here for. I can not see any problem with that. In sbt you can run `show unmanagedResourceDirectories` and you should see which directories are unmanaged resource dirs, like `conf` by default.

What I can not reproduce is that the files get "randomly" selected, and you write "most" of the files. I can not see that, maybe there is another problem with your setup or you mix the `unmanagedResourceDirectories` setting with others, like the `mappings` you describe below.

**Comment by kostafey** [comment]
Here is a workaround. To keep the required files in the `resources` folder they should be mentioned in `build.sbt`. These are actually the files that were previously listed explicitly in `build.sbt` and remained in `resources` after the build:
```scala
mappings in Universal += file("resources/someFile.xml") ->  "resources/someFile.xml"
```

**Reply by mkurz** [reply]
This is not a workaround, this is just how things work. Neither Play nor sbt nor sbt-native-packager does know anything about the `resources` folder or it's content, so with the `mapping` you tell sbt where to package a file within the distribution. Again, this is nothing special, this is just how it works.

**Comment by mkurz** [comment]
Like written, I can not reproduce a problem here and can not see what's actually wrong. All you describe is how things should work.

**Reply by kostafey** [reply]
Yes, there is actually no randomization at all. I just forgot about the `mappings` section in my `build.sbt` :). So, all the files that were not mentioned in `mappings` end up in the `conf` folder from `resources` after running `sbt dist`. But in some previous version of the Play, they just stay in `resources` as is. Should I revert and check it out again?

**Reply by mkurz** [reply]
> But in some previous version of the Play, they just stay in `resources` as is.

OK, you mean that the `resources` folder basicially were copied 1:1 into the root of the distribution package. It's possible that happend in an old version of Play, but this behaviour was definitely wrong. I tested with latest Play and can confirm how things work now is correct.
So, if you do not define any `unmanagedResourceDirectories` then a `resources` folder will not be copied into the distribution. It's like Play does not know anything about it. You need to add it with `unmanagedResourceDirectories` so it gets copied into the `conf` folder.
And if you using `mapping` you can copy it wherever you want into the distribution.

One thing didn't get mentioned here yet: In https://www.playframework.com/documentation/3.0.x/Anatomy you can see there is this special `dist` folder you can use as well:
```
dist                     → Arbitrary files to be included in your projects distribution
```
So if you create a `dist` folder in the root of your project, all it's content will be copied 1:1 into the distribution.
E.g. `dist/myfile` will end up as `myfile` in the root of your dist. So if you want to copy a `resources` folder you can add it to `dist/resources/` and it will be available as `resources` in your dist. This all can be done without configuring any `unmanagedResourceDirectories` or `mappings` Maybe that is easier for you.
